include/proxsuite/proxqp/dense/utils.hpp

```diff
@@ -239,7 +239,8 @@ global_primal_residual_infeasibility(VectorViewMut<T> ATdy,
   T bound_z = qpsettings.eps_primal_inf * infty_norm(dz.to_eigen());
 
   bool res = infty_norm(ATdy.to_eigen()) <= bound_y && eq_inf <= -bound_y &&
-             infty_norm(CTdz.to_eigen()) <= bound_z && in_inf <= -bound_z;
+             infty_norm(CTdz.to_eigen()) <= bound_z && in_inf <= -bound_z &&
+             infty_norm(dy.to_eigen()) !=0 && infty_norm(dz.to_eigen()) !=0;
```

Comment:
For efficiency reasons, I would first check `infty_norm(dy.to_eigen()) !=0 && infty_norm(dz.to_eigen()) !=0;` and return the solution if this statement if false.
@Bambade I have a question concerning the unscaling of the variables `ATdy`, `CTdy`, etc. Are they used outside of `global_primal_residual_infeasibility`?

Reply:
OK, it is better indeed.  
The unscaled ATdy, CTdy are not used outside of the function (the scaled versions are used  inside).

Reply:
I did a suggestion of how we could change it accordingly, first checking `nfty_norm(dy.to_eigen()) !=0 && infty_norm(dz.to_eigen()) !=0;`